src/MICore/Transports/PipeTransport.cs

```diff
@@ -35,6 +36,14 @@ protected override string GetThreadName()
             return "MI.PipeTransport";
         }
 
+        public override int DebuggerPid
+        {
+            get
+            {
+                return _debuggerPid;
```

Comment:
PipeTransport is also used with PipeLaunchOptions, which is the case I was thinking of. I suppose this is okay. In this case I would suggest updating the documentation for DebuggerPid to indicate that the behavior is undefined for non-local transport.